src/region/RegionAnnotations.tsx

```diff
@@ -99,39 +101,22 @@ export default class RegionAnnotations extends React.Component<Props, State> {
         this.setState({ targetRef });
     };
 
-    scaleShape({ height, width, x, y, ...rest }: Rect, invert = false): Rect {
-        const { scale } = this.props;
-        const ratio = invert ? 1 / scale : scale;
-
-        return {
-            ...rest,
-            height: height * ratio,
-            width: width * ratio,
-            x: x * ratio,
-            y: y * ratio,
-        };
-    }
-
     render(): JSX.Element {
-        const { activeAnnotationId, annotations, isCreating, staged, status } = this.props;
+        const { activeAnnotationId, annotations, isCreating, scale, staged, status } = this.props;
         const { targetRef } = this.state;
         const canDraw = !staged || !staged.message;
         const canReply = status === CreatorStatus.staged;
 
         return (
             <>
                 {/* Layer 1: Saved annotations */}
-                <svg className="ba-RegionAnnotations-list">
-                    {annotations.map(({ id, target }) => (
-                        <RegionAnnotation
-                            key={id}
-                            annotationId={id}
-                            isActive={!isCreating && activeAnnotationId === id}
-                            onSelect={this.handleAnnotationActive}
-                            shape={this.scaleShape((target as TargetRegion).shape)}
-                        />
-                    ))}
-                </svg>
+                <RegionList
+                    activeId={isCreating ? activeAnnotationId : null}
```

Comment:
Should this be `isCreating ? null : activeAnnotationId`?

Reply:
Nice catch.